File: ChristmasTreeApp/lib/models/product.dart
class Product {
  final int productId;
  final int categoryId;
  final String categoryName;
  final String name;
  final String description;
  final double price;
  final String imageUrl;
  final bool isEcoFriendly;
  final int stockQuantity;

  Product({
    required this.productId,
    required this.categoryId,
    required this.categoryName,
    required this.name,
    required this.description,
    required this.price,
    required this.imageUrl,
    required this.isEcoFriendly,
    required this.stockQuantity,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      productId: json['productId'],
      categoryId: json['categoryId'],
      categoryName: json['categoryName'],
      name: json['name'],
      description: json['description'],
      price: json['price'].toDouble(),
      imageUrl: json['imageUrl'],
      isEcoFriendly: json['isEcoFriendly'],
      stockQuantity: json['stockQuantity'],
    );
  }
}

File: ChristmasTree_API/Models/Product.cs
namespace ChristmasTreeApp.Models
{
    public class Product
    {
        public int ProductId { get; set; }
        public int CategoryId { get; set; }
        public Category Category { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public string ImageUrl { get; set; }
        public bool IsEcoFriendly { get; set; }
        public int StockQuantity { get; set; }
        public DateTime CreatedDate { get; set; }
        public ICollection<OrderItem> OrderItems { get; set; }
    }
}
